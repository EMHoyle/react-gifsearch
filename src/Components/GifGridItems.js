import React from 'react';
import PropTypes from 'prop-types';

const GifGridItems = ({title, url}) => {

    return (
        <div className="card animate__animated animate__zoomInDown">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
};

GifGridItems.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItems
