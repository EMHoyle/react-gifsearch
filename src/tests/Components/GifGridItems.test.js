import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGridItems from '../../Components/GifGridItems';

describe('GifGrid Test', () => {

    const title = 'A Title';
    const url = 'https://....';
    const wrapper = shallow( <GifGridItems title={title} url={url} /> )
    
    test('Test 1 - Enzyme: GifGrid should be render correctly', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Test 2 - Enzyme: PropTypes', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('Test 3 - Enzyme: should have a <p> with title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);

    });

    test('Test 4 - Enzyme: should have a <img> with src and alt', () => {
        
        const img = wrapper.find('img');
            expect( img.prop('src')).toBe(url);
            expect( img.prop('alt')).toBe(title);

    });

});