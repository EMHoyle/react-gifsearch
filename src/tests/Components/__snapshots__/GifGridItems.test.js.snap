// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Test Test 1 - Enzyme: GifGrid should be render correctly 1`] = `
<div
  className="card animate__animated animate__zoomInDown"
>
  <img
    alt="A Title"
    src="https://...."
  />
  <p>
    A Title
  </p>
</div>
`;

exports[`GifGrid Test Test 2 - Enzyme: PropTypes 1`] = `
<div
  className="card animate__animated animate__zoomInDown"
>
  <img
    alt="A Title"
    src="https://...."
  />
  <p>
    A Title
  </p>
</div>
`;
